audit/audit.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package audit provides an audit log writer for access to secrets.
+package audit
+
+import (
+	"encoding/json"
+	"io"
+	"net/netip"
+	"os"
+
+	"github.com/tailscale/setec/acl"
+	"github.com/tailscale/setec/types/api"
+)
+
+// Principal is the identity of a client taking action on the secrets
+// service.
+type Principal struct {
+	Hostname string     `json:"hostname"`
+	IP       netip.Addr `json:"ip"`
+	User     string     `json:"user,omitempty"`
```

Comment:
Is the idea that we will have a user OR tags?

Reply:
Yes, added documentation to the fields.